lib/galaxy/webapps/galaxy/api/notifications.py

```diff
@@ -0,0 +1,51 @@
+"""
+API operations on Notification objects.
+"""
+
+import logging
+from fastapi import (
+    Body,
+    Path,
+    Query,
+    Response,
+    status,
+)
+from galaxy.managers.notification import NotificationManager
+from galaxy.managers.context import ProvidesUserContext
+from galaxy.schema.schema import (
+    CreateNotificationPayload,
+)
+from . import (
+    depends,
+    Router,
+    DependsOnTrans
+)
+
+log = logging.getLogger(__name__)
+
+router = Router(tags=["notifications"])
+
+
+@router.cbv
+class FastAPINotifications:
+    manager: NotificationManager = depends(NotificationManager)
+
+    @router.get("/api/notifications", summary="Displays a collection (list) of notifications.")
+    def index(self):
+        return self.manager.index(limit=5)
+
+    @router.get(
+        "/api/notifications",
+        summary="Displays information about a notification.",
+    )
+    def show(self, notification_id):
+        return self.manager.show(notification_id)
+
+    @router.post("/api/notifications", summary="Create a notificaton message")
```

Comment:
You need to annotate a response model here, you can't just return the database model here. This is true for all the methods here.